src/lib/comms/include/sol-netctl.h

```diff
@@ -0,0 +1,476 @@
+    SOL_NETCTL_SERVICE_STATE_ONLINE,
+    /*
+     * @brief the service is disconnect
+     *
+     * a service is disconnect when the service is going to terminate
+     * the current connection and will return to "idle".
+     */
+    SOL_NETCTL_SERVICE_STATE_DISCONNECT,
+    /*
+     * @brief the service is failure
+     *
+     * a service is failure when the service indicates a wrong behavior.
+     */
+    SOL_NETCTL_SERVICE_STATE_FAILURE,
+    /*
+     * @brief the service is remove
+     *
+     * a service is remove when the service is not available and removed
+     * from the network system. At the same time, the service handle(pointer)
+     * will become invalid.
+     */
+    SOL_NETCTL_SERVICE_STATE_REMOVE,
+};
+
+#define SOL_NETCTL_SERVICE_TYPE_ETHERNET   "ethernet"  /**< @brief ethernet service type string */
+#define SOL_NETCTL_SERVICE_TYPE_WIFI       "wifi"      /**< @brief wifi service type string */
+#define SOL_NETCTL_SERVICE_TYPE_BLUETOOTH  "bluetooth" /**< @brief bluetooth service type string */
+#define SOL_NETCTL_SERVICE_TYPE_CELLULAR   "cellular"   /**< @brief bluetooth service type string */
+#define SOL_NETCTL_SERVICE_TYPE_GPS        "gps"       /**< @brief gps service type string */
+#define SOL_NETCTL_SERVICE_TYPE_VPN        "vpn"       /**< @brief vpn service type string */
+#define SOL_NETCTL_SERVICE_TYPE_GADGET     "gadget"    /**< @brief gadget service type string */
+#define SOL_NETCTL_SERVICE_TYPE_P2P        "p2p"       /**< @brief p2p service type string */
+
+/**
+ * @enum sol_netctl_state
+ *
+ * @brief the global connection state of system
+ *
+ * One of these must be chosen for the global connection state,
+ * with sol_netctl_get_state()
+ *
+ */
+enum sol_netctl_state {
+    /*
+     * @brief the state is unknow
+     *
+     * a state is unknow when the state is init
+     * and the state can't be given.
+     */
+    SOL_NETCTL_STATE_UNKNOWN = 0,
+    /*
+     * @brief the state is idle
+     *
+     * a state is idle when no serives is in either "ready" or "online" state.
+     */
+    SOL_NETCTL_STATE_IDLE,
+    /*
+     * @brief the state is ready
+     *
+     * a state is ready when at least one service is in "ready"
+     * state and no service is in "online" state.
+     */
+    SOL_NETCTL_STATE_READY,
+    /*
+     * @brief the state is online
+     *
+     * a state is online when at least one service is in "online".
+     */
+    SOL_NETCTL_STATE_ONLINE,
+    /*
+     * @brief the state is offline
+     *
+     * a state is offline when the device is in offline mode and
+     * the user doesn't re-enable individual technologies like
+     * WiFi and Bluetooth while in offline mode.
+     */
+    SOL_NETCTL_STATE_OFFLINE,
+};
+
+/**
+ * @brief Service monitor callback used to inform a service changed
+ *
+ * @param data the user data
+ * @param service the monitor service
+ */
+typedef void (*sol_netctl_service_monitor_cb)
+    (void *data,
+    const struct sol_netctl_service *service);
+
```

Comment:
I've commented about this before and he remembered me that this is needed because of the sol_monitors. So it is correct